### PR TITLE
fix(engine-core): log warnings instead of errors on hydration mismatch

### LINKED ISSUE #4845: Use `console.warn` instead of `console.error` for hydration mismatches
Looking in [`hydration.ts`](https://github.com/salesforce/lwc/blob/82467e330a85036c1f4253473754e5b634e92bce/packages/%40lwc/engine-core/src/framework/hydration.ts), we are inconsistent and sometimes use `console.warn` (`logWarn`) and sometimes `console.error` (`logError`) when communicating hydration mismatches.

We should probably be consistent about this and just use `logWarn` everywhere. Especially since 1) these errors are recoverable – that's the whole point of mismatch detection, i.e. to fall back to pure client-side rendering, 2) people often get confused when they see red in the console and believe a true error has occurred, and 3) it is awkward for writing unit tests, because you may have to set up spies on both `console.warn` and `console.error` if you expect the warnings.

### PR DESCRIPTION
**Pull request by cardoso**
## Details

Closes #4845

## Does this pull request introduce a breaking change?

-   😮‍💨 No, it does not introduce a breaking change.

## Does this pull request introduce an observable change?

-   🔬 Yes, it does include an observable change.

In development mode hydration mismatch will log warnings instead of errors.

### CODE REVIEW

File: packages/@lwc/integration-karma/test-hydration/mismatches/invalid-number-of-nodes/if-true/index.spec.js [hunk 1]
```diff
@@ -14,12 +14,21 @@ export default {
...
+                ],
+            });
+        } else {
+            TestUtils.expectConsoleCallsDev(consoleCalls, {
+                warn: [
+                    'Server rendered more nodes than the client.',
+                    'Hydration completed with errors.',
+                ],
+            });
+        }
```
Comment (cardoso) [inline packages/@lwc/integration-karma/test-hydration/mismatches/invalid-number-of-nodes/if-true/index.spec.js:32]: Why does the text mismatch with `DISABLE_STATIC_CONTENT_OPTIMIZATION` ? 🤔

Tested with all flags used in CI. This was the only issue.

Comment (wjhsf) [reply packages/@lwc/integration-karma/test-hydration/mismatches/invalid-number-of-nodes/if-true/index.spec.js:32]: iirc it's something to do with how we concatenate adjacent text nodes in different scenarios, particularly how we use `\u200d` (zero-width joiner) as a placeholder

File: packages/@lwc/integration-karma/test-hydration/mismatches/attrs-expression/index.spec.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ export default {
         expect(div.getAttribute('data-static')).toBe('same-value');
 
         TestUtils.expectConsoleCallsDev(consoleCalls, {
-            error: [
```
Comment (nolanlawson) [inline packages/@lwc/integration-karma/test-hydration/mismatches/attrs-expression/index.spec.js:21]: These should have `error: []` to ensure we are not logging any errors.

File: packages/@lwc/integration-karma/test-hydration/mismatches/invalid-number-of-nodes/if-true/index.spec.js (hunk 1 above)
Comment (nolanlawson) [reply packages/@lwc/integration-karma/test-hydration/mismatches/invalid-number-of-nodes/if-true/index.spec.js:32]: In the static content optimization we only call `hydrateStaticParts`, not `hydrateElement`+`hydrateText`. So you may see slightly different hydration warnings.

**Review by nolanlawson — CHANGES_REQUESTED** [review]
I'd like us to assert no-errors as mentioned, but otherwise LGTM. Thanks a lot!

**Pull request merged**